### PR TITLE
Drop the check regarding the presence of all computed partition statistics

### PR DESCRIPTION
**Pull request by findinpath**
## Description

While computing the statistics, it may very well happen that the number of partitions in the table will change (partitions get added/dropped). Avoid failing the `ANALYZE` operation because of inequality of the number of computed partition statistics vs existing table partitions.

## Additional context and related issues

In some situations it may happen that `ANALYZE table` operation is failing with the message

```
All computed statistics must be used
```

If the number of the partitions of the table changes between the time when the computing of the partitions  stats has started and when it ended, this exception is likely to appear.
Avoid the failure of the operation by dropping the verification for the  equality of the number of computed partition statistics vs existing table partitions.

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

**Review by findepi — APPROVED** [review]

**Comment by findinpath** [comment]
Please do not merge this one yet.
It is not clear yet whether the check removed is actually causing problems.

**Comment by findinpath** [comment]
I'll continue with this PR after landing https://github.com/trinodb/trino/pull/15995

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java
Comment (findinpath) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java:1482]: The PR is not accompanied by corresponding tests because I don't see a way to consistently simulating a situation where partitions get effectively removed during the computation of statistics.

**Comment by findepi** [comment]
@ebyhr ptal

**Review by ebyhr — APPROVED** [review]

**Pull request merged**
